build/build.go

```diff
@@ -595,6 +595,28 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		so.FrontendAttrs["attest:provenance"] = "mode=min,inline-only=true"
 	}
 
+	var prvmode string
+	if v, ok := opt.Attests["attest:provenance"]; ok {
+		if prvmode, err = getProvenanceMode(*v); err != nil {
+			return nil, nil, err
+		}
+	}
+	if gitprv, err := getGitProvenance(ctx, prvmode, opt.Inputs.ContextPath, opt.Inputs.DockerfilePath); err != nil {
+		return nil, nil, err
+	} else if gitprv != nil {
+		if gitprv.Source != "" {
+			so.FrontendAttrs["label:"+specs.AnnotationSource] = gitprv.Source
+			so.FrontendAttrs["vcs:source"] = gitprv.Source
+		}
+		if gitprv.Revision != "" {
+			so.FrontendAttrs["label:"+specs.AnnotationRevision] = gitprv.Revision
+			so.FrontendAttrs["vcs:revision"] = gitprv.Revision
+		}
+		if gitprv.DockerfilePath != "" {
+			so.FrontendAttrs["label:"+DockerfileLabel] = gitprv.DockerfilePath
+		}
+	}
```

Comment:
I'm not sure about this :thinking: 

I don't think setting `--provenance mode=max` should mean that we set additional labels. Labels control should be set *exclusively* when the `BUILDX_GIT_LABELS` env variable is set. Though I think `BUILDX_GIT_LABELS` probably should set the vcs opts?

I think we could refactor `getProvenanceMode` to not do the env variable lookup, we could just do it right in this function, then, if that variable is set, set the labels.

Reply:
My thought was to have the same behavior for provenance mode and BUILDX_GIT_LABELS env vars so users can still set vcs attributes. I agree that BUILDX_GIT_LABELS should just set labels. I think we should also add a deprecation notice if this env var is being used and tell user to switch to provenance.

Reply:
I think my confusion is that in `getGitProvenance` we have string values of "full" through the function. Ideally, we'd pass both the `BUILDX_GIT_LABELS` and extracted mode into `getGitProvenance`, and then parse what we want to do at the top - then we don't have to have magic strings through it.

> I think we should also add a deprecation notice if this env var is being set and tell user to switch to provenance.

Yeah, can we do this at a higher level? Otherwise, we'd get it on every call to `toSolveOpt` :scream:

Reply:
> Yeah, can we do this at a higher level? Otherwise, we'd get it on every call to `toSolveOpt` 😱

Yes should be at same level where we display the "no output" warn in BuildWithResultHandler.